### PR TITLE
Changing OnUserQuit to allow quit message modify.

### PR DESCRIPTION
**Pull request by iwalkalone69**
The purpose of this pull request is to allow modifying quit message like we can do with part message in OnUserPart event.

The change is just removing const from parameter to allow its modification in modules.

### CODE REVIEW

**Comment by SaberUK** [comment]
It would be better if an `OnUserPreQuit` event was to be introduced which is called with FIRST_MOD_RESULT. That way modules which just want to see the quit don't have to worry if a module after them in the chain has changed the message.

**Comment by iwalkalone69** [comment]
Thanks.

I will add this event and revert the change on the existing one according to you comment.

**Comment by iwalkalone69** [comment]
I've reverted changes on **OnUserQuit** event and added **OnUserPreQuit** event.

Hope it is done right. If not, tell me how should be done. Thanks.

File: include/modules.h [hunk 1]
```diff
@@ -304,6 +304,16 @@ class CoreExport Module : public classbase, public usecountbase
...
 
+	/** Called when before a user quits.
+	 * The details of the exiting user are available to you in the parameter User *user
+	 * This event is only called when the user is fully registered when they quit. To catch
+	 * raw disconnections, use the OnUserDisconnect method.
+	 * @param user The user who is quitting
+	 * @param message The user's quit message (as seen by non-opers)
+	 * @param oper_message The user's quit message (as seen by opers)
+	 */
+	virtual ModResult OnUserPreQuit(User* user, std::string &message);
```
Comment (SaberUK) [inline include/modules.h]: This should take a LocalUser not a User.

Also, the & should be against the type not against the variable name.

File: src/usermanager.cpp [hunk 2]
```diff
@@ -174,16 +174,23 @@ void UserManager::QuitUser(User* user, const std::string& quitreason, const std:
...
-
-	ServerInstance->Logs->Log("USERS", LOG_DEBUG, "QuitUser: %s=%s '%s'", user->uuid.c_str(), user->nick.c_str(), quitreason.c_str());
-	user->Write("ERROR :Closing link: (%s@%s) [%s]", user->ident.c_str(), user->GetRealHost().c_str(), operreason ? operreason->c_str() : quitreason.c_str());
-
 	std::string reason;
 	reason.assign(quitreason, 0, ServerInstance->Config->Limits.MaxQuit);
+
+	ModResult MOD_RESULT;
+	FIRST_MOD_RESULT(OnUserPreQuit, MOD_RESULT, (user, reason));
+	if (MOD_RESULT == MOD_RES_DENY) {
```
Comment (SaberUK) [inline src/usermanager.cpp]: Thereis no need to have {} on this.

**Comment by iwalkalone69** [comment]
Applied change suggested in the code commented.

**Comment by iwalkalone69** [comment]
Is everything ok now? Thanks.

**Comment by SaberUK** [comment]
Hi, i'll try to get around to merging this before the last alpha.

**Comment by iwalkalone69** [comment]
Rebased in insp3 branch (PR #1629).

**Pull request closed without merging**